docs/usage/applications.rst

```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
+is that it can be accessed during multiple stages of the connection, and it can be injected into dependencies and
+route handlers.
+
+The Application State is an instance of the :class:`.datastructures.state.State` datastructure, and it is
+accessible via the :paramref:`~.app.Litestar.state` attribute. As such it can be accessed wherever the app instance
+is accessible.
 
-The Application State is an instance of the :class:`State <.datastructures.state.State>` datastructure, and it is accessible
-via the :class:`app.state <.app.Litestar>` attribute. As such it can be accessed wherever the app instance is accessible.
+:paramref:`~.app.Litestar.state` is one of the
+:ref:`reserved keyword arguments <usage/routing/handlers:"reserved" keyword arguments>`.
 
-It's important to understand in this context that the application instance is injected into the ASGI ``scope`` mapping for
-each connection (i.e. request or websocket connection) as ``scope["app"].state``. This makes the application accessible
-wherever the scope mapping is available, e.g. in middleware, on :class:`Request <.connection.request.Request>` and
-:class:`Websocket <.connection.websocket.WebSocket>` instances (accessible as ``request.app`` / ``socket.app``), and many
+It is important to understand in this context that the application instance is injected into the ASGI ``scope`` mapping
+for each connection (i.e. request or websocket connection) as ``scope["app"].state``. This makes the application
+accessible wherever the scope mapping is available, e.g. in middleware, on :class:`~.connection.request.Request` and
+:class:`~.connection.websocket.WebSocket` instances (accessible as ``request.app`` / ``socket.app``), and many
 other places.
 
-Therefore, state offers an easy way to share contextual data between disparate parts of the application, as seen below:
+Therefore, :paramref:`~.app.Litestar.state` offers an easy way to share contextual data between disparate parts
+of the application, as seen below:
 
 .. literalinclude:: /examples/application_state/using_application_state.py
     :caption: Using Application State
-    :language: python
 
 .. _Initializing Application State:
 
 Initializing Application State
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To seed application state, you can pass a :class:`State <.datastructures.state.State>` object to the ``state`` kwarg of
-the Litestar constructor:
+To seed application state, you can pass a :class:`~.datastructures.state.State` object to the
+:paramref:`~.app.Litestar.state` parameter of the Litestar constructor:
 
 .. literalinclude:: /examples/application_state/passing_initial_state.py
     :caption: Using Application State
-    :language: python
 
+.. note:: :class:`~.datastructures.state.State` can be initialized with a :class:`dictionary <dict>`, an instance of
+    :class:`~.datastructures.state.ImmutableState` or :class:`~.datastructures.state.State`,
+    or a :class:`list` of :class:`tuples <tuple>` containing key/value pairs.
```

Comment:
I don't think this is true anymore - maybe it was for 1.x but its just typed `State | None` now.

Reply:
wdyt https://github.com/litestar-org/litestar/commit/1e5df3296abd3acd452d6e51d3006f98ac824146

Reply:
Sorry - I was wrong about this. I thought this was referring to the value that could be passed to the application on instantiation, not what can be passed to the state object on instantiation.